src/hardware/mixer.cpp

```diff
@@ -117,33 +121,69 @@ static struct mixer_t mixer = {};
 
 uint8_t MixTemp[MIXER_BUFSIZE] = {};
 
-MixerChannel::MixerChannel(MIXER_Handler _handler, const char *_name)
+MixerChannel::MixerChannel(MIXER_Handler _handler, const char *_name,
+                           const std::set<ChannelFeature> _features)
         : name(_name),
           envelope(_name),
-          handler(_handler)
+          handler(_handler),
+          features(_features)
 {}
 
+bool MixerChannel::HasFeature(ChannelFeature feature)
+{
+	return features.find(feature) != features.end();
+}
+
 bool MixerChannel::StereoLine::operator==(const StereoLine &other) const
 {
 	return left == other.left && right == other.right;
 }
 
-mixer_channel_t MIXER_AddChannel(MIXER_Handler handler, const int freq, const char *name)
+static void set_global_channel_settings(mixer_channel_t channel)
+{
+	const auto sect = static_cast<Section_prop *>(control->GetSection("mixer"));
+	assert(sect);
+
+	// Global crossfeed
+	constexpr auto default_crossfeed_strength = 0.3f;
+	auto crossfeed = 0.0f;
+
+	const std::string crossfeed_pref = sect->Get_string("crossfeed");
+	if (crossfeed_pref == "on")
+		crossfeed = default_crossfeed_strength;
+	else {
+		const auto cf = to_finite<double>(crossfeed_pref);
+		if (std::isfinite(cf) && cf >= 0.0 && cf <= 100.0) {
+			crossfeed = cf / 100.0f;
+		} else {
+			LOG_WARNING("MIXER: Invalid crossfeed value '%s', using off",
+			            crossfeed_pref.c_str());
```

Comment:
In other parts (of the Staging additions), if user content is broken, in my own additions I've tended to fallback to the out-of-the-box default, going by what I'd prefer for myself if I don't know what I'm doing (with any interface or user setting.. be it physical or software): let the designers given me what they recommend.

I'm also thinking about scenarios where people feed in non-Staging conf files or using fronends that might be feed broken values too.

That's more just a comment to chat; not asking for change.

Reply:
Yeah, I agree with that approach, and I think I'm doing exactly that here as the default crossfeed setting is `off`.

Reply:
Oh, my bad! Pebkac event here 😅 

It's this bit that had me thinking that cross feed was on by default:

```
// Global crossfeed
	constexpr auto default_crossfeed_strength = 0.3f;
	auto crossfeed = 0.0f;
```

With the local writable value the one we're populating here. My thinking was "if we can't successfully populate `crossfeed` then they get `default_crossfeed`.

But yeah, I didn't check that crossfeed as a toggle is off by default, but that the strength when-on (or auto?) Is 0.3.

I'm think moving the default only inside the `on` if-branch would have been enough to clarify this for me.